composer.json

```diff
@@ -14,24 +14,24 @@
     ],
 
     "require": {
-        "php": "^7.2 || ^8.0",
+        "php": "8.1.* || 8.2.* || 8.3.* || 8.4.* ",
```

Comment:
I've added this as per @acoulton's request, but this is not an industry standard.
Normally, this line would be contain the lowest version and above: `php: >= 8.1`
There is nothing special about this project that requires to put an upper boundary to be `8.4`

Reply:
Nobody knows what will be in PHP 8.5, so we have no idea if behat will work with it. We do know that behat required changes to work on 8.4 (and 8.3 ISTR). If we're going to advertise supported PHP versions, we should be accurate.

Reply:
Other projects are in the same boat and they do not declare such dependencies. Adding an upper bound will mean that a code change will be required in the future to allow supporting PHP 8.5. Minor PHP versions should be compatible between each other and only add deprecations. Setting `^8.1` lock the major version where we truly do not know what to expect. But it should be totally okay for minor php versions.

Reply:
> Other projects are in the same boat and they do not declare such dependencies.

IMO they should. As a user, it is much easier if composer can just tell you when/whether it's worth considering bumping a project to the next PHP than to start the process and then find problems that need to be fixed upstream before you can continue.

> Adding an upper bound will mean that a code change will be required in the future to allow supporting PHP 8.5. 

Not a code change. 

Best case, once a year we will need to add a version to the build matrix and a version to the composer.json - a very easy task, and an easy contribution for someone. If it goes green, fine. If it fails, then we know what needs fixing before it's worth people running it in their own projects.

> Minor PHP versions should be compatible between each other and only add deprecations. Setting ^8.1 lock the major version where we truly do not know what to expect.

Every PHP minor has [backwards-incompatible changes](https://www.php.net/manual/en/migration83.incompatible.php) (as well as deprecations). They're often edge cases, but there's no way to know in advance which might affect us.

Also behat - unlike some other projects - explicitly sets and runs an error_reporting level including E_DEPRECATED, so the deprecations will also break a behat build. People might have opinions on that, but changing it would be a BC break so for the 3.x series at least that will always be the case.

> But it should be totally okay for minor php versions.

The historical evidence is that assumption is false : as you can see in the CHANGELOG, PHP 8.2 was the only minor version behat was able to support without code changes.